### ISSUE TITLE
Recommend putting AZ in locality flag for cloud deployments

### ISSUE THREAD

**Issue by bdarnell** [cockroachdb/docs#4640]
If the AZ is not reflected in the --locality flag, then ranges may not be properly diversified across locality zones and will be at an increased risk of outage during an availability zone failure. We should update our production recommendations/best practices pages for cloud deployments to incorporate this:

- In most cases, for deployments to the big three clouds, the locality flag should include both the region and AZ, and look like `--locality=region=us-east-1,az=us-east-1c`. The `region` portion is optional for single-region clusters, but it doesn't hurt to always include it. Users may add additional levels to this if they wish.
- Zone configs should generally reference regions, not AZs. The user will not need to refer to the AZs anywhere outside the `--locality` flag, but it will be used by the replica placement system to ensure proper diversity.
- Different regions can have different numbers of availability zones (Most have three, but consult your cloud's documentation. For aws: https://aws.amazon.com/about-aws/global-infrastructure/). Prefer regions with at least three AZs whenever possible.

**Comment by mwang1026** [comment]
cc @piyush-singh seems server-y?

**Comment by jseldess** [comment]
Definitely. This relates to https://github.com/cockroachdb/docs/issues/10311, actually, which @taroface is taking under Deploy and Ops. @taroface, feel free to consolidate these.

**Comment by taroface** [comment]
@udnay This is consistent with your guidance on #10479, right?

**Comment by bdarnell** [comment]
> is the AZ needed by cockroachdb to make the cluster AZ aware?

Yes. The only way CockroachDB learns anything about groups of nodes is via the locality flag.

> For CRDB to make the right call do we need to specify all AZs in a region?

We don't need an exhaustive list of AZs anywhere, but every node must specify *its* AZ. (it's up to GKE to ensure that the nodes it creates are well-distributed across AZs)
